File: src/test/java/com/simprints/libsimprints/IdentificationTest.java
package com.simprints.libsimprints;

import android.os.Parcel;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import junit.framework.Assert;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;


@RunWith(AndroidJUnit4.class)
public class IdentificationTest {
    @Test
    public void testIdentificationParcelling() {
        Identification exampleId = new Identification("case-id", 99, Tier.TIER_1);

        Parcel parcel = Parcel.obtain();
        exampleId.writeToParcel(parcel, 0);

        parcel.setDataPosition(0);

        Identification createdFromParcel = Identification.CREATOR.createFromParcel(parcel);
        Assert.assertEquals(exampleId, createdFromParcel);
    }

    @Test
    public void testSorting() {
        ArrayList<Identification> idList;
        Identification topId = new Identification("top-id", 99, Tier.TIER_1);
        Identification midId1 = new Identification("first-mid-id", 51, Tier.TIER_3);
        Identification midId2 = new Identification("second-mid-id", 51, Tier.TIER_3);
        Identification lowId = new Identification("low-id", 2, Tier.TIER_5);
        Identification[] descendingOrder = new Identification[]{topId, midId1, midId2, lowId};

        Identification[] f = new Identification[]{midId1, lowId, topId, midId2};
        idList = new ArrayList<>(Arrays.asList(f));

        Collections.sort(idList);
        for (int i = 0; i < descendingOrder.length; i++) {
            Assert.assertEquals(descendingOrder[i], idList.get(i));
        }
    }
}

File: src/test/java/com/simprints/libsimprints/IdentificationTest.kt
package com.simprints.libsimprints

import androidx.core.os.bundleOf
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith


@RunWith(AndroidJUnit4::class)
class IdentificationTest {

    @Test
    fun `test identification parcelling`() {
        val expected = Identification("case-id", 99, Tier.TIER_1)
        val actual = bundleOf("test" to expected).getParcelable<Identification>("test")

        assertEquals(expected, actual)
    }

    @Test
    fun `test sorting`() {
        val topId = Identification("top-id", 99, Tier.TIER_1)
        val midId1 = Identification("first-mid-id", 51, Tier.TIER_3)
        val midId2 = Identification("second-mid-id", 51, Tier.TIER_3)
        val lowId = Identification("low-id", 2, Tier.TIER_5)

        val descendingOrder = listOf(topId, midId1, midId2, lowId)
        val sortedList = listOf(midId1, lowId, topId, midId2).sorted()

        for (i in descendingOrder.indices) {
            assertEquals(descendingOrder[i], sortedList[i])
        }
    }
}
